sphinxcontrib/datatemplates/loaders.py

```diff
@@ -18,34 +19,69 @@ def __init__(self, loader, name, match_source):
 
 
 def loader_for_source(source, default=None):
+    "Return the loader for the named source."
     for e in registered_loaders:
         if e.match_source(source):
             return e.loader
     return default
 
 
 def loader_by_name(name, default=None):
+    "Return the loader registered with the given name."
     for e in registered_loaders:
         if e.name == name:
             return e.loader
     return default
 
 
+def mimetype_loader(name, mimetype):
+    "A data loader for the exact mimetype."
+    def check_mimetype(source):
+        guess = mimetypes.guess_type(source)[0]
+        if not guess:
+            return False
+        return guess == mimetype
+    return append_loader(name, check_mimetype)
+
+
+def lenient_mimetype_loader(name, mimetype_fragment):
+    "A data loader for a mimetype containing the given substring."
+    def check_mimetype(source):
+        guess = mimetypes.guess_type(source)[0]
+        if not guess:
+            return False
+        return mimetype_fragment in guess
+    return append_loader(name, check_mimetype)
+
+
+def file_extension_loader(name, extensions):
+    "A data loader for filenames ending with one of the given extensions."
+    def check_ext(filename):
+        return pathlib.Path(filename).suffix.lower() in set(
+            e.lower() for e in extensions)
+    return append_loader(name, check_ext)
+
+
 def append_loader(name, match_source=None):
+    """Add a named loader
+
+    Add a named data loader with an optional function for matching to
+    source names.
+
+    """
     def wrap(loader_func):
         registered_loaders.append(LoaderEntry(loader_func, name, match_source))
         return loader_func
 
     return wrap
 
 
-@append_loader("csv",
-               (lambda source: mimetypes.guess_type(source)[0] == "test/csv"))
 @contextlib.contextmanager
```

Comment:
Not required but it would be nice to register this one, too.

Reply:
Done, and fixed the issue with the lookup code as a result so it was good that you pointed this out.